### DISCUSSION TITLE
Error: WebSocket connection closed abnormally with code: 3000

### DISCUSSION THREAD

**Discussion by Confucian-e** [wevm/wagmi#2827] [Q&A] [answered]
I do not use websocket but meet an error: `WebSocket connection closed abnormally with code: 3000 `
It's consoled continuously.

My Code:
```typescript
const { chains, publicClient } = configureChains(
  [mainnet],
  [
    alchemyProvider({ apiKey: " My API Key " }),
    publicProvider()
  ],
);

const { connectors } = getDefaultWallets({
  appName: 'RainbowKit App',
  projectId: 'YOUR_PROJECT_ID',
  chains,
});

const wagmiConfig = createConfig({
  autoConnect: true,
  connectors,
  publicClient,
});
```

**Comment by 0xtailf0xtail** [comment, marked as answer]
I think it's caused by `WalletConnectConnector` that's included in `getDefaultWallets`.
Please try the following for setting up `wagmiConfig`.
```
const wagmiConfig = createConfig({
  autoConnect: true,
  connectors: [
    new MetaMaskConnector({ chains }),
    new CoinbaseWalletConnector({
      chains,
      options: {
        appName: 'wagmi',
      },
    }),
    new InjectedConnector({
      chains,
      options: {
        name: 'Injected',
        shimDisconnect: true,
      },
    }),
  ],  
  publicClient,
})
```

If you need WalletConnect, you most-likely needs proper `projectId` that can be obtained from [WalletConnect Cloud](https://cloud.walletconnect.com/).
https://www.rainbowkit.com/docs/migration-guide#2-supply-a-walletconnect-cloud-projectid

**Reply by Confucian-e** [reply]
Thanks, I think I should go to ask RainbowKit.
